packages/web/resolvers/comment.ts

```diff
@@ -22,9 +22,80 @@ import {
   assignBadge,
   UserWithRels,
   ThreadWithRels,
+  sendNewBadgeEmail,
 } from './utils'
 import { NotFoundError } from './errors'
 
+const assignCommentCountBadges = async (db: PrismaClient, userId: number): Promise<void> => {
+  // Use a raw query here because we'll soon have a number of post count
+  // badges and we could end up with quite a bit of back and fourth
+  // querying, whereas here we can just make one roundtrip.
+  const newBadgeCount = await db.$executeRaw`
```

Comment:
Now that we're doing this ins two places (and we  may have more in the future) we might consider making this a little more generic. Maybe something like (and I haven't  tested this):

```ts
const joinSqlFragments = (parts: WhateverSqlType[], delimiter: WhateverSqlType) => {
  let joinedFragment = parts[0] || Prisma.empty
  for (let i = 1; i < parts.length; i++) {
    joinedFragment = Prisma.sql`${joinedFragment}\n${parts[i]}`
  }
  return joinedFragment
}

const assignCountBadges = (db: PrismaClient, userId: number, countQuery: WhateverSqlType, tiers: { [key: number]: BadgeType }) => {
  return db.$executeRaw`
    WITH count_query AS (${countQuery})
    INSERT INTO "UserBadge" ("type", "userId") (
      ${joinSqlFragments(
        Object.entries(tiers)
          .map(([threshold, badge]) => Prisma.sql`
            (
              SELECT
                ${badge}::"BadgeType" AS "type",
                ${userId}::integer AS "userId"
              FROM count_query WHERE count_query.count >= ${threshold}
            )
          `)
        )
      }
    )
    ON CONFLICT DO NOTHING
  `
}

const commentTiers = {
  10: BadgeType.TEN_COMMENTS,
  50: BadgeType.FIFTY_COMMENTS,
  ...
}

const commentCountQuery = Prisma.sql`
        SELECT COUNT(*) AS count
        FROM "Comment"
        WHERE
          "authorId" = ${userId}
`

assignCountBadges(db, userId, commentCountQuery, commentTiers)
```

where `WhateverSqlType` is whatever the type of ```Prisma.sql`` ``` is.

Typing all this out I realize it's a bit complicated so we can go over it together at some point if you want, but there's a lot of commonality here so it'd be nice to unify it. Also the `joinSqlFragments` func would be useful in the glob or raw SQL we're doing in the posts resolver too.